controllers/util/groupsnapshotclasses.go

```diff
@@ -79,3 +82,51 @@ func NewDefaultCephFsGroupSnapshotClass(
 	}
 	return gsc
 }
+
+func VolumeGroupSnapshotClassFromExisting(
+	ctx context.Context,
+	kubeClient client.Client,
+	volumeGroupSnapshotClassName string,
+	consumer *ocsv1a1.StorageConsumer,
+	consumerConfig StorageConsumerResources,
+	rbdStorageId,
+	cephFsStorageId,
+	nfsStorageId string,
+) (*groupsnapapi.VolumeGroupSnapshotClass, error) {
+	gsc := &groupsnapapi.VolumeGroupSnapshotClass{}
+	gsc.Name = volumeGroupSnapshotClassName
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(gsc), gsc); err != nil {
+		return nil, err
+	}
+	clientProfileName := ""
+	provisionerSecretName := ""
+	storageId := ""
+	operatorNamespace := consumer.Status.Client.OperatorNamespace
+	switch gsc.Driver {
+	case RbdDriverName:
+		clientProfileName = consumerConfig.GetRbdClientProfileName()
+		provisionerSecretName = consumerConfig.GetCsiRbdProvisionerSecretName()
+		storageId = rbdStorageId
+	case CephFSDriverName:
+		clientProfileName = consumerConfig.GetCephFsClientProfileName()
+		provisionerSecretName = consumerConfig.GetCsiCephFsProvisionerSecretName()
+		storageId = cephFsStorageId
```

Comment:
Is the expectation that when the user creates this groupsnapshot class, they will provide it with "pool"/"fsName" parameters? Or do we want to fill in some default values?

Reply:
> they will provide it with "pool"/"fsName" parameters
- yes, that's the expectation which is inline w/ current [documented](https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.18/html/managing_and_allocating_storage_resources/storage-classes_rhodf#creating-storage-classes-and-pools_rhodf) procedures